nodes/srl/srl.go

```diff
@@ -209,11 +209,20 @@ func (s *srl) PreDeploy(_ context.Context, params *nodes.PreDeployParams) error
 		agents := s.Cfg.Extras.SRLAgents
 
 		appmgr := filepath.Join(s.Cfg.LabDir, "config", "appmgr")
-
 		utils.CreateDirectory(appmgr, 0777)
 
+		// process extras -> agents configurations
 		for _, fullpath := range agents {
 			basename := filepath.Base(fullpath)
+			// if it is a url extract filename from url or content-disposition header
+			if utils.IsHttpUri(fullpath) {
+				basename = utils.FilenameForURL(fullpath)
+			}
+			// enforce yml extension
+			if ext := filepath.Ext(basename); !(ext == ".yml" || ext == ".yaml") {
+				basename = basename + ".yml"
+			}
```

Comment:
@steiler this is likely a simpler way?

Reply:
Rome and all the streets that lead there and so ... ;-)